pkg/handlers/apk.go

```diff
@@ -0,0 +1,403 @@
+}
+
+// createZipReader creates a new ZIP reader from the input fileReader.
+func createZipReader(input fileReader) (*zip.Reader, error) {
+	size, err := input.Size()
+	if err != nil {
+		return nil, err
+	}
+	zipReader, err := zip.NewReader(input, size)
+	if err != nil {
+		return nil, err
+	}
+	return zipReader, err
+}
+
+// parseResTable parses the resources.arsc file and returns the ResourceTable.
+func parseResTable(zipReader *zip.Reader) (*apkparser.ResourceTable, error) {
+	for _, file := range zipReader.File {
+		if file.Name == "resources.arsc" {
+			rdr, err := openFile(file)
+			if err != nil {
+				return nil, err
+			}
+
+			resTable, err := apkparser.ParseResourceTable(rdr)
+			rdr.Close()
+			if err != nil {
+				return nil, err
+			}
+			return resTable, nil
+		}
+	}
+	return nil, errors.New("resources.arsc file not found in the APK archive")
+}
+
+// openFile opens the file from the zip archive and returns the data as an io.ReadCloser
+// Note: responsibility of calling function to close the reader
+func openFile(file *zip.File) (io.ReadCloser, error) {
+	rc, err := file.Open()
+	if err != nil {
+		return nil, err
+	}
+	return rc, nil
+}
+
+// extractStringsFromResTable extracts the strings from the resources table
+// Note: This is a hacky way to get the strings from the resources table
+// APK strings are typically (always?) stored in the 0x7f000000-0x7fffffff range
+// https://chromium.googlesource.com/chromium/src/+/master/build/android/docs/life_of_a_resource.md
+func extractStringsFromResTable(resTable *apkparser.ResourceTable) (io.Reader, error) {
+	var resourceStrings bytes.Buffer
+	inStrings := false
+	for i := 0x7f000000; i <= 0x7fffffff; i++ {
+		entry, _ := resTable.GetResourceEntry(uint32(i))
+		if entry == nil {
+			continue
+		}
+		if entry.ResourceType == "string" {
+			inStrings = true
+			val, err := entry.GetValue().String()
+			if err != nil {
+				return nil, err
+			}
+			// Write directly to the buffer
+			resourceStrings.WriteString(entry.Key)
+			resourceStrings.WriteString(": ")
+			resourceStrings.WriteString(val)
+			resourceStrings.WriteString("\n")
+		}
+		// Exit the loop if we've finished processing the strings
+		if inStrings && entry.ResourceType != "string" {
+			break
+		}
+	}
+	return &resourceStrings, nil
+}
+
+// processDexFile decodes the dex file and returns the relevant instructions
+func processDexFile(ctx logContext.Context, rdr io.ReadCloser) (io.Reader, error) {
+	// dextk.Read() requires an io.ReaderAt interface
+	dexReader, err := dextk.Read(iobuf.NewBufferedReaderSeeker(rdr))
+	if err != nil {
+		return nil, err
+	}
+
+	defaultKeywords := DefaultDetectorKeywords()
+
+	// Get relevant instruction data from the dex file
+	var dexOutput bytes.Buffer
+	ci := dexReader.ClassIter()
+	for ci.HasNext() {
+		node, err := ci.Next()
+		if err != nil {
+			break
+		}
+		processDexClass(ctx, dexReader, node, defaultKeywords, &dexOutput)
+	}
+
+	return &dexOutput, nil
```

Comment:
It was intentional b/c during testing I was getting errors that didn't really impact our ability to scan data, but that would otherwise force our dex file processing to fail for a particular file. I think logging and continuing makes the most sense. I'll add that in.